packages/application-config/test/json-schema.spec.js

```diff
@@ -0,0 +1,68 @@
+const { validateConfig } = require('../src');
+const fixtureConfigSimple = require('./fixtures/config-simple.json');
+const fixtureConfigFull = require('./fixtures/config-full.json');
+const fixtureConfigEnvVariables = require('./fixtures/config-env-variables.json');
+
+describe.each`
+  name               | config
+  ${'Simple'}        | ${fixtureConfigSimple}
+  ${'Full'}          | ${fixtureConfigFull}
+  ${'Env variables'} | ${fixtureConfigEnvVariables}
+`('validating config "$name"', ({ config }) => {
+  it('should detect the config as valid', async () => {
+    await validateConfig(config);
+  });
+});
+
+describe('invalid configurations', () => {
+  it('should validate that "env" is defined', () =>
+    expect(
+      validateConfig({
+        ...fixtureConfigSimple,
+        env: undefined,
+      })
+    ).rejects.toEqual(
```

Comment:
Maybe `toMatchInlineSnapshot` is also nice here. Just wondering. I think they're a bit underused at times :)